### ISSUE TITLE
[FR] Expect redbean to support websocket

### ISSUE THREAD

**Issue by lost4git** [jart/cosmopolitan#439]

**Comment by jart** [comment]
I agree. Redbean should support websockets. It's something we don't have planned immediately but we certainly intend to do it in the not too distant future.

**Comment by javajosh** [comment]
I am also quite interested in websocket support. Any leads on this in terms of libraries? Or do you intent to implement something in the `redbean.c` file?

**Comment by jart** [comment]
We intend to implement it since the mechanics of websockets are pretty deeply intertwined with the workings of a web server. If there's any off the shelf helpers we can use for some key aspects of it, then we'd use them. But overall the process is just drilling through the RFC and writing tests.

**Comment by javajosh** [comment]
Today I've forked and built `jart/cosmopolitan`, started reading through `redbean.c` and also `rfc6455.txt` (each of which, interestingly enough, are about 7k lines). I have little experience with c, but I find the code quite readable. Justine, can you recommend any resources that might proactively ease my pain with c and its tooling, especially debugging with gdb? TIA

**Comment by jart** [comment]
Join our Discord #help channel and I'll mentor you in the art of C. Like you I have a background in Java (particularly Guava) so I could draw upon some of that experience to help clarify anything you need. https://discord.gg/FunsGsDG

**Comment by jonnybrooks** [comment]
I'd be very intertested in contributing to the implementation of WebSockets in redbean! I wrote a very simple (non spec-compliant) WebSocket server in C a couple years back and since discovering cosmo was wondering if there was something I could contribute to!

**Comment by JakeBamrah** [comment]
Forgive me if this i a naïve question here. I've been looking into this (along with [RFC6455](https://datatracker.ietf.org/doc/html/rfc6455)) for the last few days. I am a beginner C programmer but would love to give this a shot (although I think it will take a few months). Could this be broken down into two sub-task:
1. Implementing websocket api + tests
2.  _Then_ integrating it into `redbean` and `greenbean` 

or does / should these two tasks be worked on in tandem?

**Comment by pkulchenko** [comment]
@JakeBamrah, there is an implementation being worked on in #967, so you may want to review and discuss it there.
